components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -126,18 +119,37 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
                     });
 
                 _sinkBlock = new ActionBlock<NetworkMessageModel[]>(
-                    async input => await _messageSink.SendAsync(input).ConfigureAwait(false),
+                    async input => {
+                        await _messageSink.SendAsync(input).ConfigureAwait(false);
+                        if (_batchTriggerInterval > TimeSpan.Zero) {
+                            _batchTriggerIntervalTimer.Change(_batchTriggerInterval, _batchTriggerInterval);
```

Comment:
Shouldn't this be done upon sending?   And why change it to always trigger, isn't it better to only trigger once?   Then reset it when it was send to retrigger?

Reply:
send triggering is performed in case of "queue full" or "timer expired". I  wanted to avoid time triggering in case the queue is (quasi) empty right after a "queue full" trigger. The idea is to send out at least one message per batch trigger interval, but to keep them as little and full as possible. I'm stopping the timer before triggering since I don't know how long sending will take and I want to avoid double time triggering.

Reply:
done - thanks!